File: Competitive Programming 3/Section 1.4/Games (Others), Easier/12239 - Bingo!/C++/main.cpp
#include<bits/stdc++.h>
#define lim 90

using namespace std;

int arr[lim+1],N,B;
bool possible[lim];

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    while(cin>>N>>B,N){
        for(int i=0;i<B;i++)
            cin>>arr[i];
        memset(possible,0,sizeof(possible));
        int remain=N;
        for(int i=0;i<B;i++)
            for(int j=i+1;j<B;j++)
                if(!possible[abs(arr[i]-arr[j])-1]){
                    possible[abs(arr[i]-arr[j])-1]=true;
                    remain--;
                }
        cout<<(remain?'N':'Y')<<'\n';
    }
    return 0;
}

File: Competitive Programming 3/Section 1.4/Games (Others), Easier/12239 - Bingo!/Python/main.py
def main():
    while True:
        N,B=map(int,input().split())
        if N==0:
            return
        arr,possible,remain=list(map(int,input().split())),[False]*N,N
        for i in range(B):
            for j in range(i+1,B):
                if not possible[abs(arr[i]-arr[j])-1]:
                    possible[abs(arr[i]-arr[j])-1]=True
                    remain-=1
        print('N'if remain else'Y')
main()